src/graphics/gfx_screen.cpp

```diff
@@ -213,19 +252,31 @@ void DynamicScreen()
                 vScreen[2].Width = ScreenW;
                 vScreen[2].Left = 0;
                 vScreen[2].Top = 0;
-                GetvScreenAverage2();
-                if(DScreenType != 3)
-                    PlaySound(SFX_Camera);
+
+                int l, t;
+                if(g_compatibility.free_level_res)
+                {
+                    GetvScreenAverage2();
+                    l = vScreenX[1];
+                }
+                else
+                {
+                    GetvScreenAverage2Canonical(&l, &t);
+                }
+
                 for(A = 1; A <= 2; A++)
                 {
                     vScreen[A].TempDelay = 200;
                     vScreen[A].TempY = 0;
-                    vScreen[A].tempX = -vScreenX[1] + ScreenW * 0.5 - Player[A].Location.X - Player[A].Location.Width * 0.5;
+                    vScreen[A].tempX = -l + FieldW * 0.5 - Player[A].Location.X - Player[A].Location.Width * 0.5;
                 }
-                vScreen[2].Visible = true;
+
+                if(DScreenType != 3)
+                    PlaySound(SFX_Camera);
                 DScreenType = 3;
+                vScreen[2].Visible = true;
             }
-            else if(level[Player[1].Section].Height - level[Player[1].Section].Y > ScreenH && ((!vScreen[2].Visible && Player[2].Location.Y + vScreenY[1] >= ScreenH * 0.75 - vScreenYOffset - Player[2].Location.Height) || (vScreen[2].Visible && Player[2].Location.Y + vScreenY[1] >= ScreenH * 0.75 - vScreenYOffset - Player[2].Location.Height)) && (Player[1].Location.Y < level[Player[1].Section].Height - ScreenH * 0.75 - vScreenYOffset - Player[1].Location.Height))
+            else if(level[Player[1].Section].Height - level[Player[1].Section].Y > FieldH && ((!vScreen[2].Visible && Player[2].Location.Y + vScreenY[1] >= FieldH * 0.75 - vScreenYOffset - Player[2].Location.Height) || (vScreen[2].Visible && Player[2].Location.Y + vScreenY[1] >= FieldH * 0.75 - vScreenYOffset - Player[2].Location.Height)) && (Player[1].Location.Y < level[Player[1].Section].Height - FieldH * 0.75 - vScreenYOffset - Player[1].Location.Height))
```

Comment:
So messy conditions suggested to be split into multiple lines, making less pain for eyes :eyes: